test/helpers/customError.js

```diff
@@ -1,19 +1,37 @@
 const { config } = require('hardhat');
+const { expect } = require('chai');
 
 const optimizationsEnabled = config.solidity.compilers.some(c => c.settings.optimizer.enabled);
 
 /** Revert handler that supports custom errors. */
-async function expectRevertCustomError(promise, reason) {
+async function expectRevertCustomError(promise, expectedErrorName, args) {
```

Comment:
For errors with no argument, that would avoid having to specify the empty array.
```suggestion
async function expectRevertCustomError(promise, expectedErrorName, args = []) {
```

Reply:
This was intentional because I thought the number of arguments should be explicit in the tests. Open to discussion anyway but I'll leave it until the end because it requires fixing the errors with `[]`

Reply:
For me if I see `expectRevertCustomError(..., "AnError")` I will assume that this assertion is ignoring the arguments. I think the array makes it explicit that there are zero arguments.

I think I would keep the array explicit, but consider that we may want to reevaluate this in the future if we find it annoying to work with.

Reply:
Do we want to change the helper so that if `args == null` the arguments check is skipped ?

Reply:
> Do we want to change the helper so that if args == null the arguments check is skipped ?

I don't think so